src/librbd/CMakeLists.txt

```diff
@@ -15,6 +15,8 @@ endif()
 
 add_library(rbd_types STATIC
   ${librbd_types_srcs})
+target_link_libraries(rbd_types PRIVATE
```

Comment:
We're getting link errors otherwise, same for the other libraries:

```
[100%] Linking CXX executable ../../../bin/rbd.exe
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x9d3): undefined reference to `cls::rbd::SnapshotNamespace::encode(ceph::buffer::v15_2_0::list&) const'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0xb26): undefined reference to `cls::rbd::SnapshotNamespace::encode(ceph::buffer::v15_2_0::list&) const'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0xd16): undefined reference to `cls::rbd::SnapshotNamespace::encode(ceph::buffer::v15_2_0::list&) const'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x1656): undefined reference to `cls::rbd::SnapshotNamespace::dump(ceph::Formatter*) const'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x1706): undefined reference to `cls::rbd::SnapshotNamespace::dump(ceph::Formatter*) const'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x1868): undefined reference to `cls::rbd::SnapshotNamespace::decode(ceph::buffer::v15_2_0::list::iterator_impl<true>&)'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x1a98): undefined reference to `cls::rbd::SnapshotNamespace::decode(ceph::buffer::v15_2_0::list::iterator_impl<true>&)'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x1b48): undefined reference to `cls::rbd::SnapshotNamespace::decode(ceph::buffer::v15_2_0::list::iterator_impl<true>&)'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x1b68): undefined reference to `cls::rbd::SnapshotNamespace::decode(ceph::buffer::v15_2_0::list::iterator_impl<true>&)'
/usr/bin/x86_64-w64-mingw32-ld: ../../../lib/librbd_types.a(WatchNotifyTypes.cc.obj):WatchNotifyTypes.cc:(.text+0x13a8): undefined reference to `cls::rbd::SnapshotNamespace::dump(ceph::Formatter*) const'
```

Reply:
Hmm -- I wonder if that's just a library order issue (some linkers care more than others). i.e. so long as librbd_types.a is specified before cls_rbd_client.a, it should work.

Reply:
My bad, the library ordering is indeed the problem here. I'll update the PR.